src/main/java/org/jabref/gui/entryeditor/fileannotationtab/FulltextSearchResultsTab.java

```diff
@@ -1,37 +1,66 @@
 package org.jabref.gui.entryeditor.fileannotationtab;
 
+import java.io.IOException;
 import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
-import javafx.scene.web.WebView;
+import javafx.geometry.Orientation;
+import javafx.scene.control.ContextMenu;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.Separator;
+import javafx.scene.control.Tooltip;
+import javafx.scene.input.MouseButton;
+import javafx.scene.text.Text;
+import javafx.scene.text.TextFlow;
 
+import org.jabref.gui.DialogService;
 import org.jabref.gui.StateManager;
+import org.jabref.gui.actions.ActionFactory;
+import org.jabref.gui.actions.StandardActions;
+import org.jabref.gui.desktop.JabRefDesktop;
 import org.jabref.gui.entryeditor.EntryEditorTab;
-import org.jabref.gui.util.OpenHyperlinksInExternalBrowser;
-import org.jabref.gui.util.Theme;
+import org.jabref.gui.maintable.OpenExternalFileAction;
+import org.jabref.gui.maintable.OpenFolderAction;
 import org.jabref.logic.l10n.Localization;
 import org.jabref.model.entry.BibEntry;
 import org.jabref.model.entry.LinkedFile;
 import org.jabref.model.pdf.search.PdfSearchResults;
 import org.jabref.model.pdf.search.SearchResult;
 import org.jabref.model.search.rules.SearchRules;
-import org.jabref.preferences.FilePreferences;
+import org.jabref.preferences.PreferencesService;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class FulltextSearchResultsTab extends EntryEditorTab {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(FulltextSearchResultsTab.class);
+
     private final StateManager stateManager;
-    private final FilePreferences filePreferences;
+    private final PreferencesService preferencesService;
+    private final DialogService dialogService;
+    private final ActionFactory actionFactory;
 
-    private final WebView webView;
+    private final ScrollPane scrollPane;
+    private final TextFlow content;
 
-    public FulltextSearchResultsTab(StateManager stateManager, Theme theme, FilePreferences filePreferences) {
+    private BibEntry entry;
+
+    public FulltextSearchResultsTab(StateManager stateManager, PreferencesService preferencesService, DialogService dialogService) {
```

Comment:
Please stay at `FilePreferences` if possible

Reply:
What about the StandardAction.OPEN_FOLDER and OPEN_EXTERNAL_FILE? They need the PreferencesService.